internal/utils/fs.go

```diff
@@ -38,3 +39,30 @@ func Move(oldPath, newPath string) error {
 
 	return nil
 }
+
+// ExistsAndIsWritable checks if the directory specified by the path exists and is writable.
+func ExistsAndIsWritable(path string) error {
+	info, err := os.Stat(path)
+	if errors.Is(err, os.ErrNotExist) {
```

Comment:
```suggestion
	if errors.Is(err, errors.Is(err, fs.ErrNotExist)) {
```
We should use the `errors` package.

Reply:
Ignore this suggestion, it's already doing what I meant. My bad!